components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -567,14 +567,31 @@ protected void concludeFlow(HttpServletRequest request, HttpServletResponse resp
                     if (StringUtils.isNotBlank(authHistory.getIdpSessionIndex()) &&
                             StringUtils.isNotBlank(authHistory.getIdpName())) {
                         try {
-                            if (!userSessionStore.hasExistingFederatedAuthSession(authHistory.getIdpSessionIndex())) {
-                                userSessionStore.storeFederatedAuthSessionInfo(sessionContextKey, authHistory);
+                            if (FrameworkUtils.isTenantIdColumnAvailableInFedAuthTable()) {
+                                int tenantId = IdentityTenantUtil.getTenantId(context.getTenantDomain());
+                                if (!userSessionStore.isExistingFederatedAuthSessionAvailable(
+                                        authHistory.getIdpSessionIndex(), tenantId)) {
+                                    userSessionStore.storeFederatedAuthSessionInfo(sessionContextKey, authHistory,
+                                            tenantId);
+                                } else {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug(String.format("Federated auth session with the id: %s already " +
+                                                "exists", authHistory.getIdpSessionIndex()));
+                                    }
+                                    userSessionStore.updateFederatedAuthSessionInfo(sessionContextKey, authHistory,
+                                            tenantId);
+                                }
                             } else {
-                                if (log.isDebugEnabled()) {
-                                    log.debug(String.format("Federated auth session with the id: %s already exists",
-                                            authHistory.getIdpSessionIndex()));
+                                if (!userSessionStore.hasExistingFederatedAuthSession(
+                                        authHistory.getIdpSessionIndex())) {
+                                    userSessionStore.storeFederatedAuthSessionInfo(sessionContextKey, authHistory);
+                                } else {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug(String.format("Federated auth session with the id: %s already " +
+                                                "exists", authHistory.getIdpSessionIndex()));
```

Comment:
add a space after "exists"

Reply:
The comma after the log message needs to be in the next immediate place without a space. Missed a full stop and added.